lib/livebook_web/live/session_live.ex

```diff
@@ -2272,10 +2272,16 @@ defmodule LivebookWeb.SessionLive do
   end
 
   defp is_secret_on_session?(secret, secrets) do
-    Map.has_key?(secrets, secret.name)
+    Map.has_key?(secrets, secret.name) && secrets[secret.name] == secret.value
```

Comment:
if both sides are boolean we should make that explicit by using ` and ` over ` && `. In any case this is equivalent to shorter:

```suggestion
    Map.get(secrets, secret.name) == secret.value
```